ludwig/trainers/trainer.py

```diff
@@ -884,8 +884,21 @@ def train(
                     if self.distributed.is_model_parallel():
                         # Assume the full weights cannot fit in memory on GPU
                         self.model = self.model.cpu()
-                    _, unexpected_keys = self.model.load_state_dict(state_dict, strict=False)
-                    assert unexpected_keys == [], f"Unexpected keys found in state dict: {unexpected_keys}"
+
+                    # For a full explanation of this 8-bit workaround, see https://github.com/ludwig-ai/ludwig/pull/3606
+                    # TODO (jeffkinnison): Investigate why `self.model` is seemingly not placed on GPU before this point
+                    if hasattr(self.model.config_obj, "quantization") and self.model.config_obj.quantization.bits == 8:
+                        if torch.cuda.is_available():
+                            self.model.model.cuda()
+                            self.model.model.cpu()
+                        _, unexpected_keys = self.model.load_state_dict(state_dict, strict=False)
+                        only_weights_format_keys = ["weights_format" in k for k in unexpected_keys]
+                        assert (
+                            unexpected_keys == [] or only_weights_format_keys
+                        ), f"Unexpected keys found in state dict: {unexpected_keys}"
```

Comment:
Add something about the `only_weights_format_keys` to the error message.

Reply:
Added in [b8b487d](https://github.com/ludwig-ai/ludwig/pull/3606/commits/b8b487d10d15a81fa3f68a5631e56880ad4f09f5)

Reply:
Hmm, am I missing something, I still don't see anything in the assert message about it?